docs/benchmarking-workflow-corefx.md

```diff
@@ -0,0 +1,253 @@
+
+Every time you want to run the benchmarks against local build of CoreFX you need to provide the path to CoreRun:
+
+```cmd
+dotnet run -f netcoreapp3.0 --coreRun "C:\Projects\corefx\artifacts\bin\runtime\netcoreapp-Windows_NT-Release-x64\CoreRun.exe" --filter $someFilter 
+```
+
+**Note:** BenchmarkDotNet expects a path to `CoreRun.exe` file (`corerun` on Unix), not to `Core_Root` folder.
+
+Once you rebuild the part of CoreFX you are working on, the appropriate `.dll` gets updated and the next time you run the benchmarks CoreRun is going to load the updated library.
+
+```cmd
+C:\Projects\corefx\src\System.Text.RegularExpressions\src> dotnet msbuild /p:ConfigurationGroup=Release
+```
+
+## Preventing Regressions
+
+Preventing regressions is one of the fundamental concepts of our performance culture.
+
+**Before introducing any changes**, you should run the benchmarks that test the performance of the feature that you are going to work on and store the results in a **dedicated** folder.
+
+```cmd
+C:\Projects\performance\src\benchmarks\micro> dotnet run -f netcoreapp3.0 \
+    --artifacts "C:\results\before" \
+    --coreRun "C:\Projects\corefx\artifacts\bin\runtime\netcoreapp-Windows_NT-Release-x64\CoreRun.exe" \
+    --filter System.IO.Pipes*
+```
+
+Please try to **avoid running any resource-heavy processes** that could **spoil** the benchmark results while running the benchmarks.
+
+You can also create a **copy** of the folder with CoreRun and all the libraries to be able to run the benchmarks against the **unmodified base** in the future.
+
+After you introduce the changes and rebuild the part of CoreFX that you are working on **in Release** you should re-run the benchmarks. Remember to store the results in a different folder.
+
+```cmd
+C:\Projects\corefx\src\System.IO.Pipes\src> dotnet msbuild /p:ConfigurationGroup=Release
+
+C:\Projects\performance\src\benchmarks\micro> dotnet run -f netcoreapp3.0 \
+    --artifacts "C:\results\after" \
+    --coreRun "C:\Projects\corefx\artifacts\bin\runtime\netcoreapp-Windows_NT-Release-x64\CoreRun.exe" \
+    --filter System.IO.Pipes*
+```
+
+When you have the results you should use [ResultsComparer](../src/tools/ResultsComparer/README.md) to find out how your changes have affected the performance:
+
+```cmd
+C:\Projects\performance\src\tools\ResultsComparer> dotnet run --base "C:\results\before" --diff "C:\results\after" --threshold 5%
```

Comment:
Can you elaborate what threshold/noise means?
`threshold for Statistical Test. Examples: 5%, 10ms, 100ns, 1s`

Why pick 5%?

Reply:
> Can you elaborate what threshold/noise means?

This is explained in the referenced doc:

> --noise - noise threshold for Statistical Test. The difference for 1.0ns and 1.1ns is 10%, but it's just a noise. Examples: 0.5ns 1ns. The default value is 0.3ns.

So I would say that `--noise` is additional threshold-based filter to filter out the noise for nano-benchmarks. Do you think that this description is enough?

> Why pick 5%?

It's just a sample value, but it's most probably too high. I am going to lower it to 2%

Reply:
> Do you think that this description is enough?

I get what noise means. What does threshold mean?

Reply:
It's a value that sets the limit for the statistical test, which answers the question if there is a regression or not.

`--threshold 5%`  means show me all the regressions and improvements bigger than 5%

`--threshold 10ms`  means show me all the regressions and improvements bigger than 10 miliseconds

Reply:
@ahsonkhan I am going to mark this comment as "resolved", please feel free to ask more questions